src/main/java/org/opensearch/flowframework/util/JsonToJsonTransformer.java

```diff
@@ -0,0 +1,388 @@
+
+        if (shouldBeArray || extracted.size() > 1) {
+            ArrayNode arrayNode = MAPPER.createArrayNode();
+            for (Object value : extracted) {
+                arrayNode.add(MAPPER.valueToTree(value));
+            }
+            outputNode.set(propertyKey, arrayNode);
+        } else {
+            outputNode.set(propertyKey, MAPPER.valueToTree(extracted.get(0)));
+        }
+    }
+
+    /**
+     * Handles array properties (keys containing [*] or [index]).
+     *
+     * TERMINATION CONDITIONS:
+     * - Wildcard arrays: bounded by determineArraySize() result
+     * - Explicit index arrays: processes single specified index
+     * - Empty size results in no processing
+     *
+     * @param outputNode the target object
+     * @param arrayPropertyKey the array property key (e.g., "items[*]")
+     * @param elementMapping the mapping for array elements
+     * @param inputDocument the source document
+     */
+    private static void processArrayProperty(
+        ObjectNode outputNode,
+        String arrayPropertyKey,
+        JsonNode elementMapping,
+        Object inputDocument
+    ) {
+        ArrayKeyInfo keyInfo = parseArrayKey(arrayPropertyKey);
+        ArrayNode arrayNode = outputNode.withArray(keyInfo.propertyName);
+
+        if (keyInfo.isWildcard) {
+            // Wildcard array - size determined by actual data
+            int arraySize = determineArraySize(elementMapping, inputDocument);
+
+            for (int i = 0; i < arraySize; i++) {
+                ObjectNode elementNode = MAPPER.createObjectNode();
+                arrayNode.add(elementNode);
+
+                // Build array element with wildcard substitution
+                JsonNode processedMapping = substituteWildcardsInMapping(elementMapping, i);
+                buildOutputFromMapping(elementNode, processedMapping, inputDocument);
+            }
+        } else {
+            // Explicit index - process single element
+            int targetIndex = keyInfo.explicitIndex;
+
+            // Extend array to accommodate index
+            while (arrayNode.size() <= targetIndex) {
+                arrayNode.add(MAPPER.createObjectNode());
+            }
+
+            ObjectNode elementNode = (ObjectNode) arrayNode.get(targetIndex);
+
+            // Build array element without substitution (explicit index paths already have concrete indices)
+            buildOutputFromMapping(elementNode, elementMapping, inputDocument);
+        }
+    }
+
+    /**
+     * Determines if a property key represents an array (contains [ and ]).
+     *
+     * TERMINATION: Simple string analysis, no recursion.
+     */
+    private static boolean isArrayProperty(String key) {
+        return key.contains("[") && key.contains("]");
+    }
+
+    /**
+     * Parses array key into components.
+     *
+     * TERMINATION: Simple parsing logic, no recursion.
+     */
+    private static ArrayKeyInfo parseArrayKey(String arrayKey) {
```

Comment:
This manual reconstruction of only some JsonPath features seems concerning.  Doesn't the JsonPath library already include all possible parsing per its spec in a `JsonPath.parse()` method?  Look at the MLInference processor implementations [like this](https://github.com/opensearch-project/ml-commons/blob/a29750f15b6175fe298ec6528babf1251fd795d1/plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchRequestProcessor.java#L269-L271)